src/goto-instrument/code_contracts.cpp

```diff
@@ -163,6 +163,41 @@ bool code_contractst::has_contract(const irep_idt fun_name)
   return type.has_contract();
 }
 
+void code_contractst::add_quantified_variable(
+  exprt expression,
+  replace_symbolt &replace,
+  irep_idt mode)
+{
+  // If the expression is a quantified expression, this function adds
+  // the quantified variable to the symbol table and to the expression map
+
+  // TODO Currently only works if the contract contains only a single
+  // quantified formula
+  // i.e. (1) the top-level element is a quantifier formula
+  // and (2) there are no inner quantifier formulae
+  if(expression.id() == ID_exists || expression.id() == ID_forall)
```

Comment:
Could we also turn this into an invariant instead of an if statement (e.g., `INVARIANT(expression.id() == ID_exists || expression.id() == ID_forall)`)? I think this function will only be called for these two expressions.

Reply:
I think using an INVARIANT would restrict this function too much, especially down the line, when this function will become recursive. In my current design, I call this function on every function contract, regardless whether they contain a quantified expression or not. If no quantified expression is found, then this function simply does nothing. However, having an INVARIANT would cause the program to fail if no quantifier is found, which is not what desired behavioor, in my opinion.